eth/client.go

```diff
@@ -162,46 +161,35 @@ func NewClient(accountAddr ethcommon.Address, keystoreDir string, eth *ethclient
 	if err != nil {
 		return nil, err
 	}
+	backend.SetMaxGasPrice(maxGasPrice)
 
 	am, err := NewAccountManager(accountAddr, keystoreDir, signer)
 	if err != nil {
 		return nil, err
 	}
 
-	return &client{
+	if err := am.Unlock(password); err != nil {
+		return nil, err
+	}
+
+	// Setting gasLimit and gasPrice to zero, nil allows it to be overwritten by eth.Backend
+	opts, err := am.CreateTransactOpts(0, nil)
```

Comment:
Looks like a change was added here after the squash and before the merge to master. 

- We don't need this if we're calling SetGasInfo
- setContracts is called once in this constructor and then again in SetGasInfo